lib/amazonka/src/Amazonka/Auth/ConfigFile.hs

```diff
@@ -89,10 +115,19 @@ fromFilePath profile credentialsFile configFile env = do
           Just (cp, mRegion) -> do
             env' <- case cp of
               ExplicitKeys authEnv ->
-                pure env {_envAuth = Identity $ Auth authEnv}
+                pure env {envAuth = Identity $ Auth authEnv}
               AssumeRoleFromProfile roleArn sourceProfileName -> do
-                sourceEnv <- evalConfig config sourceProfileName
-                fromAssumedRole roleArn "amazonka-assumed-role" sourceEnv
+                seenProfiles <- lift get
+                if sourceProfileName `elem` seenProfiles
+                  then
+                    liftIO
+                      . Exception.throwIO
+                      . SourceProfileLoop
+                      $ reverse seenProfiles
```

Comment:
Nice!

Reply:
I rewrote this part since I don't think it needs a new exception constructor, but yea it wasn't too bad.